src/PrestaShopBundle/Translation/Exporter/ThemeExporter.php

```diff
@@ -184,68 +192,80 @@ protected function ensureFileBelongsToExportDirectory($filePath)
         $validFileLocation = substr(realpath($filePath), 0, strlen(realpath($this->exportDir))) === realpath($this->exportDir);
 
         if (!$validFileLocation) {
-            throw new \Exception('Invalid file location. This file should belong to the export directory');
+            throw new Exception('Invalid file location. This file should belong to the export directory');
         }
 
-        return $validFileLocation;
+        return true;
     }
 
     /**
      * @param string $themeName
      * @param string $locale
-     * @param bool $rootDir
+     * @param string|bool $rootDir
      *
-     * @return \Symfony\Component\Translation\MessageCatalogue
+     * @return MessageCatalogue
+     *
+     * @throws FileNotFoundException|PrestaShopException
      */
-    protected function getCatalogueExtractedFromTemplates($themeName, $locale, $rootDir = false)
-    {
-        $tmpFolderPath = $this->getTemporaryExtractionFolder($themeName);
-        $folderPath = $this->getFlattenizationFolder($themeName);
+    protected function getCatalogueExtractedFromTemplates(
+        string $themeName,
+        string $locale,
+        $rootDir = false
+    ): MessageCatalogue {
+        $theme = $this->themeRepository->getInstanceByName($themeName);
 
-        $this->filesystem->remove($folderPath);
-        $this->filesystem->remove($tmpFolderPath);
+        $cachedFilesPath = $this->themeExtractor->getCachedFilesPath($theme);
+        $temporaryFilesPath = $this->themeExtractor->getTemporaryFilesPath($theme);
 
-        $this->filesystem->mkdir($folderPath);
-        $this->filesystem->mkdir($tmpFolderPath);
+        $this->filesystem->remove($cachedFilesPath);
+        $this->filesystem->remove($temporaryFilesPath);
 
-        $theme = $this->themeRepository->getInstanceByName($themeName);
-        $this->themeExtractor
-            ->setOutputPath($tmpFolderPath)
+        $tmpExtractPath = $temporaryFilesPath . DIRECTORY_SEPARATOR . $locale;
+
+        $this->filesystem->mkdir($tmpExtractPath);
```

Comment:
Symfony Filesystem `mkdir()` ignores already existing directories. Could it become an issue, if same function is called twice in a row ? What about the usecase where 2nd call is triggered 1st call has finished (an angry user double-clicks the export button) ?

Also new directory is created (by default) with 0700 rights. Shouldn't it be 0755 ?

Reply:
The 2nd call will find the directory empty and an empty catalogue will be returned until the cache is refreshed. The only solution I have a in mind to reduce the risk is to check if a file is in the directory IF it exists before

For the rights 0700 means the web user will have all the rights on the directory. I don't see in which case 'groups' and 'others' need the right to read+execute but I can add them ;)